### ISSUE TITLE
Fine-tuning finished in 1.5 hours

### ISSUE THREAD

**Issue by ZhengTang1120** [tatsu-lab/stanford_alpaca#168]
Hello,

Thanks for sharing this amazing work!

I tried to fine-tune Alpaca-7b. I used the same data in this repo and the same command posted in the readme file. The only different thing I did is instead of installing transformers using the fork you mentioned in read me, I used the main version from huggingface, which they support LlamaTokenizer, LlamaForCausalLM now.

The question is I finished the fine-tuning in 1.5 hours on 4 A100 80G GPUs, but in your blog, you mentioned that it took 3 hours to fine tune on 8 A100 GPUs. I am wondering what caused this difference. Any ideas? Should I worried about the quality of my fine-tuned model?

Thank you!
Zheng Tang

**Comment by jeffwadsworth** [comment]
Do you mind revealing what your train loss and validation accuracy percentages were in the end?

**Comment by ZhengTang1120** [comment]
No problem.
This is the last training log: {'train_runtime': 4744.911, 'train_samples_per_second': 32.879, 'train_steps_per_second': 0.257, 'train_loss': 0.7364038264222921, 'epoch': 3.0}
I didn't see any validation accuracy in the log file.

**Comment by ZhengTang1120** [comment]
[train-alpaca-log.txt](https://github.com/tatsu-lab/stanford_alpaca/files/11125595/train-alpaca-log.txt)
And this is all the losses in the log.

**Comment by nfsp3k** [comment]
Similar issue here.
For me, it took around 40 minutes with 8 80GB A100 GPUs.
The last train loss is around 0.5 and the trained model performs bad. (it cannot handle any instructions appropriately like the original LLaMA.)

**Comment by Hiusam** [comment]
I train on 8 80G A100 with 2 batchsize per device and 3 epochs. The last train loss is:

```
    {
      "epoch": 3.0,
      "step": 1218,
      "total_flos": 1.2039681644848742e+17,
      "train_loss": 0.7313933935511876,
      "train_runtime": 3166.1436,
      "train_samples_per_second": 49.273,
      "train_steps_per_second": 0.385
    }
  ],
  "max_steps": 1218,
  "num_train_epochs": 3,
  "total_flos": 1.2039681644848742e+17,
  "trial_name": null,
  "trial_params": null
```

And it can generate responses properly.
```
Prompt: Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
The highest mountain in China is 

### Response:
Result:   The highest mountain in China is Mount Everest, which is located on the border of Tibet and Qinghai provinces and stands at an elevation of 8,848 meters (29,029 feet).</s>
--------------------------------------------------
Prompt: Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
1+1=?

### Response:
Result:   2</s>
--------------------------------------------------
Prompt: Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Who are you?

### Response:
Result:   I'm a 21-year-old student studying computer science. I'm passionate about coding, design, and music. I'm also a big advocate for gender equality and mental health awareness.</s>
--------------------------------------------------
Prompt: Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
Hello, have a nice day!

### Response:
Result:   Bye!</s>
```

**Comment by jianjiangant** [comment]
I fine tune  the 7b alpaca on 8 A800 with 4 batchsize and 3 epoch without offload like this :

torchrun --nproc_per_node=8 --master_port=11011 train.py \
    --model_name_or_path /work/llama-7b-hf  \
    --data_path ./alpaca_data.json \
    --bf16 True \
    --output_dir /work/fine-tuned-model-diff-2  \
    --num_train_epochs 3 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 2000 \
    --save_total_limit 1 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --fsdp "full_shard auto_wrap" \
    --fsdp_transformer_layer_cls_to_wrap 'LlamaDecoderLayer' \
    --tf32 True

it takes about 40min.  Below is some info of my trainer_state:
 {
      "epoch": 3.0,
      "step": 1218,
      "total_flos": 1.473143831003136e+17,
      "train_loss": 0.7372715054034013,
      "train_runtime": 2432.1592,
      "train_samples_per_second": 64.143,
      "train_steps_per_second": 0.501
  }

then I use weight_diff.py to recover the weight with path_raw llama-7b-hf，but the inference result is so bad.

python weight_diff.py recover --path_raw /work/llama-7b-hf --path_diff /work/fine-tuned-model-diff-2/  --path_tuned /work/alpaca-7b-recover/test2/ 
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:08<00:00,  4.10s/it]Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:12<00:00,  4.27s/it]Using pad_token, but it is not set yet.
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 323/323 [00:04<00:00, 76.66it/s]Input: Below is an instruction that describes a task. Write a response that appropriately completes the request.

### Instruction:
List three technologies that make life easier.

### Response:
Completion: ingraozссdueposelyereateemsnouria credilleszoniss Volk *–jkulla einzazonzz MartíagioussellPGillesuten KindisaUGwo Рес Image równitzMetaillaмах‘misǐNGescimeseremAPPкуoggleinciвCTurnibőlbahieULarktwär Alanht rolloorerosyiRuntimeirasOFinholegdekstrijrog lykitermet́linerieefepurentes ranrasaticesдраandUsersRESinit Def


why the Response so strange?

**Comment by xlim1997** [comment]
Hello,
Do you mind providing your training script?
